x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Schema, SchemaConflicts } from '../../../shared/types';
+
+import { SearchSettings } from './types';
+
+interface RelevanceTuningProps {
+  searchSettings: SearchSettings;
+  schema: Schema;
+  schemaConflicts: SchemaConflicts;
+}
+
+interface RelevanceTuningActions {
+  onInitializeRelevanceTuning(props: RelevanceTuningProps): RelevanceTuningProps;
+  setSearchSettings(searchSettings: SearchSettings): { searchSettings: SearchSettings };
+  setFilterValue(value: string): string;
+  setSearchQuery(value: string): string;
+  setSearchResults(searchResults: object[]): object[];
+  setResultsLoading(resultsLoading: boolean): boolean;
+  clearSearchResults(): void;
+  resetSearchSettingsState(): void;
+  dismissSchemaConflictCallout(): void;
+}
+
+interface RelevanceTuningValues {
+  searchSettings: Partial<SearchSettings>;
+  schema: Schema;
+  dataLoading: boolean;
+  schemaConflicts: SchemaConflicts;
+  unsavedChanges: boolean;
+  filterInputValue: string;
+  query: string;
+  searchResults: object[] | null;
+  resultsLoading: boolean;
+  showSchemaConflictCallout: boolean;
+  engineHasSchemaFields: boolean;
+  schemaFields: string[];
+  schemaFieldsWithConflicts: string[];
+  filteredSchemaFields: string[];
+  filteredSchemaFieldsWithConflicts: string[];
```

Comment:
[super optional] If possible it might be nice to organize this a bit by usage (same for the reducer order below), e.g.

```js
  dataLoading: boolean;
  unsavedChanges: boolean;
  searchSettings: Partial<SearchSettings>;
  schema: Schema;
  engineHasSchemaFields: boolean;
  schemaFields: string[];
  schemaFieldsWithConflicts: string[];
  filterInputValue: string;
  filteredSchemaFields: string[];
  filteredSchemaFieldsWithConflicts: string[];
  schemaConflicts: SchemaConflicts;
  showSchemaConflictCallout: boolean;
  query: string;
  searchResults: object[] | null;
  resultsLoading: boolean;
```

If that order doesn't make sense in your head like it does in mine and my brain is weird, then no worries at all 🧠

Reply:
I can do that, as long as you don't mind a lengthly diff.

Reply:
https://github.com/elastic/kibana/pull/89313/commits/c69b184a5f24aa2eabb52ea5f45e14fa33761a17